pkg/tsdb/loki/api.go

```diff
@@ -211,12 +217,12 @@ func (api *LokiAPI) DataQuery(ctx context.Context, query lokiQuery, responseOpts
 		span.SetStatus(codes.Error, err.Error())
 		instrumentation.UpdatePluginParsingResponseDurationSeconds(ctx, time.Since(start), "error")
 		api.log.Error("Error parsing response from loki", "error", res.Error, "metricDataplane", responseOpts.metricDataplane, "duration", time.Since(start), "stage", stageParseResponse)
-		return nil, res.Error
+		return res
```

Comment:
Don't we need to track this as an datasource error as well?

Reply:
it starts with if `res.Error != nil` , so it means that the response already has an error and we are returning it.

Reply:
The main change in this cases is that we are no longer returning `err` , and instead of that we are returning response that has an `error` in it.

Reply:
How will those errors be flagged? As `downstream` or `plugin`? Or anything else?

Reply:
Or to be specific, in this case it is going to be `plugin` - as `downstream` can come only from Loki db. And we returned that before. So here if`converter.ReadPrometheusStyleResult` returns res.Error, it is going to be plugin error.